tests/Doctrine/Tests/DBAL/DBALExceptionTest.php

```diff
@@ -17,6 +17,14 @@ public function testDriverExceptionDuringQueryAcceptsBinaryData()
         $e = DBALException::driverExceptionDuringQuery($driver, new \Exception, '', array('ABC', chr(128)));
         self::assertContains('with params ["ABC", "\x80"]', $e->getMessage());
     }
+    
+    public function testDriverExceptionDuringQueryAcceptsResource()
+    {
+        /* @var $driver Driver */
+        $driver = $this->createMock(Driver::class);
+        $e = \Doctrine\DBAL\DBALException::driverExceptionDuringQuery($driver, new \Exception, "INSERT INTO file (`content`) VALUES (?)", [1 => tmpfile()]);
```

Comment:
I'd rather use something that doesn't pollute your environment each run, i.e. `fopen(__FILE__, 'r')`.

Reply:
Changed